aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
-.. include:: /../common/stub-topic.txt
+Formatting Response Data
+========================
 
-|stub-icon| Formatting
-======================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+ASP.NET Core MVC has built-in support for formatting response data, using fixed formats or in response to client specifications.
+
+.. contents:: Sections
+    :local:
+    :depth: 1
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/master/mvc/models/formatting/sample>`_.
+
+Format-Specific Action Results
+------------------------------
+
+Some ``ActionResult`` options are specific to a particular format, such as ``JsonResult`` and ``ContentResult`` (note string return types are simply formatted as text). Actions can return specific results that are always formatted in a particular manner. For example, returning a ``JsonResult`` will return JSON-formatted data, regardless of client preferences. Likewise, returning a ``ContentResult`` will return plain text formatted string data.
```

Comment:
@danroth27  kind of implies you need to select the action return type on what you are returning. Would it be worth adding this note:

An action isn't required to return any particular type; MVC supports any object return value. If users choose to return an IActionResult implementation, they have many to choose from and Controller has helper methods corresponding to many of the choices. If they choose to return anything else, the object will be serialized using the appropriate IOutputFormatter implementation.